go/libraries/doltcore/sqle/procedures_table.go

```diff
@@ -94,6 +101,88 @@ func DoltProceduresGetOrCreateTable(ctx *sql.Context, db Database) (*WritableDol
 	return tbl.(*WritableDoltTable), nil
 }
 
+// migrateDoltProceduresSchema migrates the dolt_procedures system table from a previous schema version to the current
+// schema version by adding any columns that do not exist.
+func migrateDoltProceduresSchema(ctx *sql.Context, db Database, oldTable *WritableDoltTable) (newTable *WritableDoltTable, rerr error) {
+	// Copy all the old data
+	iter, err := SqlTableToRowIter(ctx, oldTable.DoltTable, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	nameIdx := oldTable.sqlSchema().IndexOfColName(doltdb.ProceduresTableNameCol)
+	createStatementIdx := oldTable.sqlSchema().IndexOfColName(doltdb.ProceduresTableCreateStmtCol)
+	createdAtIdx := oldTable.sqlSchema().IndexOfColName(doltdb.ProceduresTableCreatedAtCol)
+	modifiedAtIdx := oldTable.sqlSchema().IndexOfColName(doltdb.ProceduresTableModifiedAtCol)
+	sqlModeIdx := oldTable.sqlSchema().IndexOfColName(doltdb.ProceduresTableSqlModeCol)
+
+	defer func(iter sql.RowIter, ctx *sql.Context) {
+		err := iter.Close(ctx)
+		if err != nil && rerr == nil {
+			rerr = err
+		}
+	}(iter, ctx)
+
+	var newRows []sql.Row
+	for {
+		sqlRow, err := iter.Next(ctx)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		newRow := make(sql.Row, ProceduresTableSchema().GetAllCols().Size())
+		newRow[0] = sqlRow[nameIdx]
+		newRow[1] = sqlRow[createStatementIdx]
+		newRow[2] = sqlRow[createdAtIdx]
+		newRow[3] = sqlRow[modifiedAtIdx]
+		if sqlModeIdx >= 0 {
+			newRow[4] = sqlRow[sqlModeIdx]
```

Comment:
Don't we want an explicit default for this column if the old values didn't have it?

Reply:
Yeah, that's probably a good call. I was assuming the caller of this function would need to handle the default, but you're right that it'd be much cleaner to just apply it here.